projects/testing-library/src/lib/models.ts

```diff
@@ -78,6 +78,27 @@ export interface RenderResult<ComponentType, WrapperType = ComponentType> extend
   renderDeferBlock: (deferBlockState: DeferBlockState, deferBlockIndex?: number) => Promise<void>;
 }
 
+declare const ALIASED_INPUT_BRAND: unique symbol;
+export type AliasedInput<T> = T & {
+  [ALIASED_INPUT_BRAND]: T;
+};
+export type AliasedInputs = Record<string, AliasedInput<unknown>>;
+
+export type ComponentInput<T> =
+  | {
+      [P in keyof T]?: T[P] extends Signal<infer U> ? U : T[P];
+    }
+  | AliasedInputs;
+
+/**
+ * @description
+ * Creates an aliased input branded type with a value
+ *
+ */
+export function aliasedInputWithValue<T>(value: T): AliasedInput<T> {
```

Comment:
The only thing that's left is to name this.
`aliasedInputWithValue` covers the load but it's a mouthful, I wonder if we can name this better. `AliasedInputs` is a good name for the type though.

Some variations:

- `aliasedInput`
- `aliasedValue`
- `aliasedInputValue`
- `inputValue`
- `aliasValue`
- `inputAliasValue`

Reply:
An alternative idea:

```
inputs: {
 myName: "Bob",
 ...aliasedInput("alias", "value")
}
```

Where `aliasedInput` simply returns basically `Record<string, AliasedInput<T>>`

Reply:
It took me a minute, but I think that might be the most straightforward to teach/use .
It also made me think about adding an `aliasInputs` property next to `inputs`, but ... 🤷


Do you have a preference?

Reply:
I'm also taking a look if we somehow access the metadata of an input field, e.g. to get the alias name.

Reply:
On a similar note, would have been awesome if `input.required()` returned some other type than `input`. 

We could then enforce required inputs to be provided in `inputs` otherwise there's a compile time error.

This can be a good PR upstream in angular.

Reply:
I was thinking the same thing! Unfortunately it seems that this isn't possible (for now).
I suppose that plugin can detect it using the AST, but that goes a step too far 😅

Reply:
@andreialecu do you have a preference about `aliasedInputWithValue` or `aliasedInput` ?

Reply:
The shorter, the better. If you like the `aliasedInput`, I have no objections :)

Reply:
> I was thinking the same thing! Unfortunately it seems that this isn't possible (for now). I suppose that plugin can detect it using the AST, but that goes a step too far 😅

PS: I opened this feature request on Angular's side:
https://github.com/angular/angular/issues/57225

It includes a TS playground link that shows how it would work.

Reply:
> The shorter, the better. If you like `aliasedInput`, I have no objections :)

I also like it because it makes it more explicit and there's a clear difference between normal and alias props.
So if you agree, and you have the time, you can update the PR if you want.

Reply:
> An alternative idea:
> 
> ```
> inputs: {
>  myName: "Bob",
>  ...aliasedInput("alias", "value")
> }
> ```
> 
> Where `aliasedInput` simply returns basically `Record<string, AliasedInput<T>>`

To clarify, you're talking about this type of usage, correct?

Reply:
@andreialecu correct

Reply:
Updated in [`539d20b` (#473)](https://github.com/testing-library/angular-testing-library/pull/473/commits/539d20b382307d1a68c6828f5a9a0ccf2f2874ae)